docs/source/developers/python.rst

```diff
@@ -558,6 +558,7 @@ Now, we can build pyarrow:
 
    $ pushd arrow\python
    $ set PYARROW_WITH_PARQUET=1
+   $ set CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1
```

Comment:
It might be a bit confusing to set it here, because that seems to indicate it is needed to _build_ pyarrow, while it is actually need to _run_ (eg running the tests or in a conole), I think?

Reply:
Not sure, when running locally I have always set it before building pyarrow, just to be sure =) Can test and change +1

Reply:
If we are going to change the page to say Python 3.10 soon, then it maybe doesn't matter too much (since then we will remove this line anyway)